### ISSUE TITLE
Can't get Firefox headless in full screen

### ISSUE THREAD

**Issue by heitorsilva** [Codeception/CodeceptJS#1929]
#### What are you trying to achieve?

I want to run CodeceptJS with Webdriver, using Chrome and Firefox, with the resolution of 1920x1200.

#### What do you get instead?

I was able to configure Chrome for what I needed, but Firefox always stays with a weird resolution. When I get the screen captures, the height is always small.

My tests are failing on Firefox because I can't get it to run on fullscreen.

### Details

* CodeceptJS version: 2.3.2
* NodeJS version: 10.16.3
* Operating System: Debian 9.11 (of a custom Docker image made with node:lts-slim)
* WebDriverIO version: 5.14.3
* Selenium Standalone Service version: 5.13.2
* Firefox version: 69.0.2
* Geckodriver version: 0.25.0
* OpenJDK version: 11.0.4

I tried three configurations and all of them fail:

```json
{
  "tests": "../src/**/*_test.js",
  "output": "../output",
  "helpers": {
    "WebDriver": {
      "url": "https://my-secret-domain.example.org",
      "browser": "firefox",
      "desiredCapabilities": {
        "moz:firefoxOptions": {
          "args": [
            "--headless",
            "--width=1920",
            "--height=1200"
          ]
        }
      },
      "windowSize": "maximize",
      "waitForTimeout": 5000,
      "smartWait": 5000,
      "restart": true
    }
  },
  "include": {
    "I": "../src/steps_file.js"
  },
  "bootstrap": null,
  "mocha": {},
  "name": "marvin",
  "plugins": {
    "wdio": {
      "enabled": true,
      "services": ["selenium-standalone"]
    }
  }
}
```

```json
{
  "tests": "../src/**/*_test.js",
  "output": "../output",
  "helpers": {
    "WebDriver": {
      "url": "https://my-secret-domain.example.org",
      "browser": "firefox",
      "desiredCapabilities": {
        "moz:firefoxOptions": {
          "args": [
            "--headless",
            "--window-size=1920,1200"
          ]
        }
      },
      "windowSize": "maximize",
      "waitForTimeout": 5000,
      "smartWait": 5000,
      "restart": true
    }
  },
  "include": {
    "I": "../src/steps_file.js"
  },
  "bootstrap": null,
  "mocha": {},
  "name": "marvin",
  "plugins": {
    "wdio": {
      "enabled": true,
      "services": ["selenium-standalone"]
    }
  }
}
```

```json
{
  "tests": "../src/**/*_test.js",
  "output": "../output",
  "helpers": {
    "WebDriver": {
      "url": "https://my-secret-domain.example.org",
      "browser": "firefox",
      "desiredCapabilities": {
        "moz:firefoxOptions": {
          "args": [
            "--headless"
          ]
        }
      },
      "windowSize": "1920x1200",
      "waitForTimeout": 5000,
      "smartWait": 5000,
      "restart": true
    }
  },
  "include": {
    "I": "../src/steps_file.js"
  },
  "bootstrap": null,
  "mocha": {},
  "name": "marvin",
  "plugins": {
    "wdio": {
      "enabled": true,
      "services": ["selenium-standalone"]
    }
  }
}
```

Here is a screenshot of what Firefox displays when the test fails:

![login_and_logout_works failed](https://user-images.githubusercontent.com/878699/66292462-a7634880-e8e4-11e9-99ac-44308fcdbdd7.png)

**Comment by PeterNgTr** [comment]
Hi @heitorsilva I just did the same on my local, it's okay :)

**Comment by heitorsilva** [comment]
Hi @PeterNgTr 

But which config did you use? And did you use the same versions as I did, for everything?

**Comment by PeterNgTr** [comment]
my versions:
```
    "codeceptjs": "2.3.2",
    "selenium-standalone": "^6.16.0",
    "webdriverio": "5.14.4",
    Node: 12.4.0 - /usr/local/bin/node
    npm: 6.9.0 - /usr/local/bin/npm
   Firefox: 69.0.2
```

**Comment by heitorsilva** [comment]
@PeterNgTr 
and which config?

```js
{
      "desiredCapabilities": {
        "moz:firefoxOptions": {
          "args": [
            "--headless"
          ]
        }
      },
      "windowSize": "1920x1200",
}
```
or
```js
{
      "desiredCapabilities": {
        "moz:firefoxOptions": {
          "args": [
            "--headless",
            "--window-size=1920,1200"
          ]
        }
      },
      "windowSize": "maximize",
}
```
or
```js
{
      "desiredCapabilities": {
        "moz:firefoxOptions": {
          "args": [
            "--headless",
            "--width=1920",
            "--height=1200"
          ]
        }
      },
      "windowSize": "maximize",
}
```

?

**Comment by PeterNgTr** [comment]
I'm using this
```js
{
      "desiredCapabilities": {
        "moz:firefoxOptions": {
          "args": [
            "--headless",
            "--width=1920",
            "--height=1200"
          ]
        }
      },
      "windowSize": "maximize",
}
```

**Comment by heitorsilva** [comment]
@PeterNgTr 
I noticed that you are using different packages than I am.

Here is my package.json:

```json
{
  "dependencies": {
    "@wdio/browserstack-service": "^5.13.2",
    "@wdio/cli": "^5.14.3",
    "@wdio/selenium-standalone-service": "^5.13.2",
    "codeceptjs": "^2.3.2"
  }
}
```

The error message I get is this:
```bash
Element <a> could not be scrolled into view
```

And the screenshot saved when it fails has the resolution of 1340x1952.

**Comment by heitorsilva** [comment]
by enabling [trace logs](https://firefox-source-docs.mozilla.org/testing/geckodriver/TraceLogs.html), I learned that:

```json
{
      "browser": "firefox",
      "desiredCapabilities": {
        "moz:firefoxOptions": {
          "args": [
            "--headless",
            "--width=1920",
            "--height=1200"
          ]
        }
      },
      "windowSize": "1920x1200",
      "logLevel": "silent"
}
```

is the right configuration for the tools I'm using.

if I leave `windowSize` with `1920x1200` without passing the `width` and `height` params, the viewport in the end gets a weird resolution.

also, if I pass the `width` and `height` correctly, but `windowSize` with `maximize`, the final resolution is 1366x768, not respecting the 1920x1200 I passed on the `width` and `height` args.

I'll close the issue.

Thanks for the support.
